### PR TITLE
WSTEAM1-1453  Update imports to resolve all import/no-relative-packages errors

### PR DESCRIPTION
**Pull request by Louis-Matsika**
Resolves JIRA [[1453](https://jira.dev.bbc.co.uk/browse/WSTEAM1-1453)]

Overall changes
======
Update imports to fix all occurences of `Relative import from another package is not allowed` error to help unblock  [Upgrade eslint-config-airbnb to 19.0.4](https://github.com/bbc/simorgh/pull/12206)

list of files that need to be updated as listed in Jira: [paper doc](https://paper.dropbox.com/doc/files-to-fix--Cb9yLl1Tpxu147SqXL~nO1MGAg-olRz5l2ONc4AlC0iHvKAu)

Code changes
======
for each file listed:
updated import statments to use [nextjs module aliases](https://nextjs.org/docs/14/pages/building-your-application/configuring/absolute-imports-and-module-aliases) `#` Instead of  relative pathing `../../../`

our current aliases are:
```
      ['#app', './src/app'],
      ['#contexts', './src/app/contexts'],
      ['#components', './src/app/legacy/components'],
      ['#containers', './src/app/legacy/containers'],
      ['#data', './data'],
      ['#hooks', './src/app/hooks'],
      ['#lib', './src/app/lib'],
      ['#psammead', './src/app/legacy/psammead'],
      ['#models', './src/app/models'],
      ['#pages', './src/app/pages'],
      ['#testHelpers', './src/testHelpers'],
      ['#server', './src/server'],
      ['#storybook', './.storybook'],
```

A  new Alias: `['#src', './src']` has been created that should allow access to every file in `/src`. allowing for some paths that previously couldn’t be updated to now be amended e.g `/src/testHelpers`

A  new Alias: `['#next', './ws-nextjs-app']` has been created that should allow access to every file in `/ws-nextjs-app`. allowing for some paths that previously couldn’t be updated to now be amended e.g `/ws-nextjs-app/pages/[service]/live/[id]/Post/types`

Testing
======
see [github checks on this commit](https://github.com/bbc/simorgh/actions/runs/12277526136/job/34257105857?pr=12224), expand the linting checks & search for `Relative import from another package is not allowed.` no results should come up as these errors should be fixed now

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Nextjs module aliases docs](https://nextjs.org/docs/14/pages/building-your-application/configuring/absolute-imports-and-module-aliases)

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/app/routes/index.test.jsx [hunk 1]
```diff
@@ -21,13 +21,14 @@ import mediaAssetPageJson from '#data/yoruba/cpsAssets/media-23256797.json';
 import { ERROR_PAGE, FRONT_PAGE } from '#app/routes/utils/pageTypes';
 import * as fetchDataFromBFF from '#app/routes/utils/fetchDataFromBFF';
 import gahuzaOnDemandAudio from '#data/gahuza/bbc_gahuza_radio/p02pcb5c.json';
+// eslint-disable-next-line import/order
+import routes from '.';
```
Comment (Louis-Matsika) [inline src/app/routes/index.test.jsx]: `import routes from '.';` seems to need to be above the alias imports from below, however the linter doesn't like this.

To get both the linter passing and path aliasing to work I had to use this command `// eslint-disable-next-line import/order`

I looked if there are other instances of this exact command in Simorgh, there isn't. but there are many instances of commands similar that start with `eslint-disable-next-line…`

There is however an instance of a command to disable the import order for a whole file `/* eslint-disable import/order */` in `src/server/utilities/customMetrics/index.test.ts`

The imports from `#app` and `#psammead` in this file specifically didn’t get flagged for needing an alias in Karina's PR #12206  that shows where errors for `Relative import from another package is not allowed.` are. if we don’t want to use this `// eslint-disable-next-line import/order `workaround we i don’t think we have to. Just thought I’d change them to use aliases anyway.

We could leave these imports as they were previously and I think they still might not show up when linting, as they weren't flagged before?

Don’t know how people feel about this?

File: src/app/lib/utilities/getBrandedImage/index.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import { getEnvConfig } from '../getEnvConfig';
+import { getEnvConfig } from '#app/lib/utilities/getEnvConfig';
+//  import { getEnvConfig } from '../getEnvConfig';
```
Comment (karinathomasbbc) [inline src/app/lib/utilities/getBrandedImage/index.js]:
```suggestion
```

File: src/app/routes/index.test.jsx [hunk 3]
```diff
@@ -19,15 +19,16 @@ import sportArticlePageJson from '#data/sport/judo/articles/cj80n66ddnko.json';
 import mediaAssetPageJson from '#data/yoruba/cpsAssets/media-23256797.json';
 
 import { ERROR_PAGE, FRONT_PAGE } from '#app/routes/utils/pageTypes';
+import * as fetchDataFromBFF from '#app/routes/utils/fetchDataFromBFF';
 import gahuzaOnDemandAudio from '#data/gahuza/bbc_gahuza_radio/p02pcb5c.json';
+// eslint-disable-next-line import/order
```
Comment (karinathomasbbc) [inline src/app/routes/index.test.jsx:24]: Can we fix the import order rather than disabling the lint rule?

Comment (Louis-Matsika) [reply src/app/routes/index.test.jsx:24]: I tried this in an [earlier commit](https://github.com/bbc/simorgh/pull/12224/commits/e10ee36141a2ebe5cc913afd029de8e0bf0aeb76), but github tests failed. Similar solutions of [rearranging imports worked in other files](https://github.com/bbc/simorgh/pull/12224/commits/4110e8089891a42a6f1db9d9dba703d07fe4b616). But this file doesn’t like it

Comment (karinathomasbbc) [reply src/app/routes/index.test.jsx:24]: Apologies, I didn't see your comment when I added this review. I'm OK with this - it's only a test file, and I'm pleased to see that the other imports are now using aliases (as it means that if we copy this file / look at it for inspiration, we would follow the same patterns).

**Comment by karinathomasbbc** [comment]
I have confirmed that upgrading eslint-config-airbnb to 19.0.4 no longer contains any `import/no-relative-packages` - see https://github.com/bbc/simorgh/actions/runs/12277526136/job/34257105857?pr=12224, expand the linting & search for import/no-relative-packages

Screenshot 2024-12-11 at 13 51 47

I will now revert [374e0da](https://github.com/bbc/simorgh/pull/12224/commits/374e0daa926302c05710bb38e6df783649edc31f)

File: src/app/routes/index.test.jsx (hunk 3 above)
Comment (Louis-Matsika) [reply src/app/routes/index.test.jsx:24]: All good. and again we don’t have to include these specific import statements to complete the work described by the Jira ticket. There are other import statements that still use relative pathing aswell, but these import statements also weren’t flagged by linting checks for `eslint-config-airbnb 19.0.4.`

Comment (karinathomasbbc) [reply src/app/routes/index.test.jsx:24]: That's interesting - I wonder why they are not included in the linting checks. It's not something to worry about for now - we just want to do the bare minimum to allow us to upgrade the eslint-config-airbnb library. Thanks for the update!

**Review by karinathomasbbc — APPROVED** [review]

**Review by emilysaffron — APPROVED** [review]

**Pull request merged**